src/uct/api/uct.h

```diff
@@ -287,30 +288,57 @@ enum uct_am_flags {
 
 
 /**
- * @ingroup UCT_AM
- * @brief Active message callback flags.
+ * @ingroup UCT_RESOURCE
+ * @brief Callback flags.
  *
- * List of flags for active message callback
+ * List of flags for a callback
  * A callback must have either SYNC or ASYNC flags.
  */
-enum uct_am_cb_flags {
-    UCT_AM_CB_FLAG_SYNC  = UCS_BIT(1), /**< Callback is always invoked from the context (thread, process)
-                                            that called uct_iface_progress(). An interface must
-                                            have @ref UCT_IFACE_FLAG_AM_CB_SYNC flag set to support sync
-                                            callback invocation */
-
-    UCT_AM_CB_FLAG_ASYNC = UCS_BIT(2)  /**< Callback may be invoked from any context. For example,
-                                            it may be called from transport async progress thread. To guarantee
-                                            async invocation, interface must have @ref UCT_IFACE_FLAG_AM_CB_ASYNC
-                                            flag set.
-                                            If async callback is set on interface with only @ref
-                                            UCT_IFACE_FLAG_AM_CB_SYNC flags, it will behave exactly like a
-                                            sync callback  */
+enum uct_cb_flags {
+    UCT_CB_FLAG_SYNC  = UCS_BIT(1), /**< Callback is always invoked from the context (thread, process)
+                                         that called uct_iface_progress(). An interface must
+                                         have @ref UCT_IFACE_FLAG_CB_SYNC flag set to support sync
+                                         callback invocation */
+
+    UCT_CB_FLAG_ASYNC = UCS_BIT(2)  /**< Callback may be invoked from any context. For example,
+                                         it may be called from transport async progress thread. To guarantee
+                                         async invocation, interface must have @ref UCT_IFACE_FLAG_CB_ASYNC
+                                         flag set.
+                                         If async callback is set on interface with only @ref
+                                         UCT_IFACE_FLAG_CB_SYNC flags, it will behave exactly like a
+                                         sync callback  */
```

Comment:
Suggest the following re-wording for clarity:
it may be called from a transport async progress thread. To guarantee async invocation, the interface must have the @ref UCT_IFACE_FLAG_CB_ASYNC flag set. If async callback is requested on an interface which only supports sync callback (i.e., only the @ref UCT_IFACE_FLAG_CB_SYNC flag is set), it will behave exactly like a sync callback.